docs/reference/builder.md

```diff
@@ -152,18 +152,22 @@ Example (parsed representation is displayed after the `#`):
     ADD . $foo       # ADD . /bar
     COPY \$foo /quux # COPY $foo /quux
 
-The instructions that handle environment variables in the `Dockerfile` are:
+Environment variables are supported by the following list of instructions in 
+the `Dockerfile`:
 
-* `ENV`
 * `ADD`
 * `COPY`
-* `WORKDIR`
+* `ENV`
 * `EXPOSE`
-* `VOLUME`
+* `ONBUILD` (see note)
 * `USER`
+* `WORKDIR`
+* `VOLUME`
 
-`ONBUILD` instructions are **NOT** supported for environment replacement, even
-the instructions above.
+> **Note**:
+> prior to 1.4, `ONBUILD` instructions did **NOT** support environment 
+> variable, even when combined with any of the instructions listed above.
+> This limitation was later fixed.
```

Comment:
Minor nit:
Could you change "fixed" to "removed"? (I think it was a deliberate decision to not support it at first, so not a "fix"